lib/rules/declaration-property-unit-allowed-list/index.js

```diff
@@ -49,6 +62,12 @@ function rule(list) {
 					return false;
 				}
 
+				const type = node.type;
+
+				if (optionsMatches(options, 'ignore', 'inside-function') && type === 'function') {
+					return false;
+				}
+
```

Comment:
[suggestion] How about this refactoring to avoid the duplicate check `node.type === 'function'`?
The diff lines increase a bit, but the logic seems more readable to me.

```js
if (node.type === 'function') {
	// Ignore wrong units within `url` function
	if (node.value.toLowerCase() === 'url') {
		return false;
	}

	if (optionsMatches(options, 'ignore', 'inside-function')) {
		return false;
	}
}
```